pkg/cli/predict.go

```diff
@@ -67,10 +72,79 @@ the prediction on that.`,
 	cmd.Flags().StringVarP(&outPath, "output", "o", "", "Output path")
 	cmd.Flags().StringArrayVarP(&envFlags, "env", "e", []string{}, "Environment variables, in the form name=value")
 	cmd.Flags().BoolVar(&useReplicateAPIToken, "use-replicate-token", false, "Pass REPLICATE_API_TOKEN from local environment into the model context")
+	cmd.Flags().StringVar(&inputJSON, "json", "", "Pass inputs as JSON object, read from file (@inputs.json) or via stdin (@-)")
 
 	return cmd
 }
 
+func parseJSONInput(jsonInput string) (map[string]any, error) {
+	var jsonStr string
+
+	if strings.HasPrefix(jsonInput, "@") {
+		// Read from file or stdin
+		source := jsonInput[1:]
+
+		if source == "-" {
```

Comment:
Same as previous comment, this shouldn't need a `@` prefix, but fine if it's supported as well.